src/game/interactables_2/sky_canyon/whirlwind.c

```diff
@@ -0,0 +1,662 @@
+
+static const u16 gUnknown_080E0124[3][3] = {
+    { SA2_ANIM_588, 2, 1 },
+    { SA2_ANIM_588, 1, 1 },
+    { SA2_ANIM_588, 0, 2 },
+};
+
+static const u8 gUnknown_080E0136[8] = { 0, 0, 0, 0, 1, 1, 1, 2 };
+
+#define UNKNOWN_WHIRLWIND_THIRYTWO 32
+
+void initSprite_Interactable_SkyCanyon_Whirlwind(MapEntity *me, u16 spriteRegionX,
+                                                 u16 spriteRegionY, u8 spriteY, u32 kind)
+{
+    struct Task *t = TaskCreate(Task_807D06C, sizeof(Sprite_IA86), 0x2010, 0,
+                                TaskDestructor_Interactable086);
+    Sprite_IA86 *ia086 = TaskGetStructPtr(t);
+    s32 someX, someY;
+    s32 value;
+    u32 temp;
+    ia086->unk228.posX = TO_WORLD_POS(me->x, spriteRegionX);
+    ia086->unk228.posY = TO_WORLD_POS(me->y, spriteRegionY);
+    ia086->unk228.kind = kind;
+    ia086->unk228.offsetX = me->d.sData[0] * TILE_WIDTH;
+    ia086->unk228.offsetY = me->d.sData[1] * TILE_WIDTH;
+    ia086->unk228.width = ia086->unk228.offsetX + me->d.uData[2] * TILE_WIDTH;
+    ia086->unk228.height = ia086->unk228.offsetY + me->d.uData[3] * TILE_WIDTH;
+
+    ia086->unk228.someX
+        = ia086->unk228.posX + ((ia086->unk228.width + ia086->unk228.offsetX) >> 1);
+
+    ia086->unk228.someY = ia086->unk228.posY + (ia086->unk228.height);
+
+    ia086->unk228.unk1C = ia086->unk228.height - ia086->unk228.offsetY;
+    ia086->unk228.unk1E = ia086->unk228.width - ia086->unk228.offsetX;
+    temp = ia086->unk228.unk1E;
+    ia086->unk228.unk20 = temp >> 2;
+    ia086->unk228.me = me;
+
+    ia086->unk228.spriteX = me->x;
+    ia086->unk228.spriteY = spriteY;
+
+    {
+        u8 i;
+        u8 *vram;
+        Unk_IA86 *unkPtr;
+        for (i = 0; i < ARRAY_COUNT(ia086->unk0); i++) {
+            ia086->unk0[i].s = NULL;
+        }
+        ia086->unk228.vramMem = VramMalloc(4);
+        vram = ia086->unk228.vramMem;
+
+        for (i = 0; i < ARRAY_COUNT(ia086->sprites); i++) {
+            Sprite *s = &ia086->sprites[i];
+            s->graphics.size = 0;
+            s->unk14 = 0;
+            s->unk1C = 0;
+            s->unk21 = 0xFF;
+            s->unk22 = 0x10;
+            s->focused = 0;
+            s->unk28->unk0 = -1;
+            s->graphics.dest = vram;
+            s->graphics.anim = gUnknown_080E0124[i][0];
+            s->variant = gUnknown_080E0124[i][1];
+            vram += gUnknown_080E0124[i][2] * 32;
+        }
+    }
+
+    sub_807D1BC(ia086);
+    SET_MAP_ENTITY_INITIALIZED(me);
+}
+
+#ifdef NON_MATCHING
+void sub_807C9C0(Sprite_IA86 *ia086)
+{
+    Player *p;
+    gPlayer.moveState |= MOVESTATE_400000;
+    gPlayer.unk64 = 44;
+
+    p = &gPlayer;
+    ia086->unk182 = 64;
+    ia086->unk184 = Q_24_8(0.5);
+    ia086->unk186 = gPlayer.speedAirY;
+    ia086->unk180 = 0;
+    ia086->unk188 = Q_24_8(ia086->unk228.someX) - gPlayer.x;
+    ia086->unk18C = Q_24_8(ia086->unk228.someY) - gPlayer.y;
+    ia086->unk190 = 0;
+    ia086->unk194 = 0;
+
+    gCurTask->main = Task_807D0C4;
+}
+#else
```

Comment:
I think my changes were undone from last night. You should remove the `NON_MATCHING` blocks from this second function